pages/instructorEffectiveUser/instructorEffectiveUser.js

```diff
@@ -44,18 +44,27 @@ router.get('/', function (req, res, next) {
     if (res.locals.ipaddress.substr(0, 7) === '::ffff:') {
       res.locals.ipaddress = res.locals.ipaddress.substr(7);
     }
+
+    // This page can be mounted under `/pl/course/...`, in which case we won't
+    // have a course instance to get a display timezone from. In that case, we'll
+    // fall back to the course, and then to `UTC`. Note that in some other places
+    // we fall back to `America/Chicago` if neither the course nor the course
```

Comment:
I think defaulting to `America/Chicago` made sense when PL was UIUC only, but now I imagine that using UTC is a better choice. I don't know how safe it is to change this at the moment. Maybe the right thing to do is to not even have a default and error out during sync if at least a course timezone isn't specified? We'd have to make that change carefully, of course.

Reply:
I think the ultimate fallback should be to `institutions.display_timezone`, which is NOT NULL. My original intention here was to always use:
```
coalesce(course_instance.display_timezone, course.display_timezone, institution.display_timezone)
```

In retrospect I think this is a bad design and we should change all three tables to have the `display_timezone` column be NOT NULL. At sync time we should fill in the value from either JSON (if provided) or from the "parent" table.

Reply:
`display_timezone` for institutions is currently a PT concept; `database/tables/institutions.pg` doesn't contain it. Even then, the default institution is set to `America/Chicago`, which would be confusing to anyone not in the Midwest. Fixing that is of course outside the scope of this PR, and I agree that defaulting to `institution.display_timezone` is ultimately the right thing to do. We should just be more cognizant of being, uh, non-Illinois-centric when picking defaults for things in the future.

All I have to do is add a migration that adds `institutions.display_timezone` such that the definition matches that from PT, correct?

Reply:
Ah, excellent point that PL doesn't know about `institutions.display_timezone` 😅 

Just make sure the migration uses `ADD COLUMN IF NOT EXISTS`.

I fully agree that we want to get away from defaulting to `America/Chicago`. I'm not sure what the best way is to manage this for courses in the default institution? Error during sync if a course doesn't have a timezone set and is in the default institution?

Reply:
The problem is that _all_ courses are in the default institution during local dev 😓 I can't see a way to make this check consistent between local dev and prod if "is timezone required for course" is dependent upon state that's not in the course repo.

Reply:
Dang, good point. Next idea:
* When we create new courses (copied from the template course), set the timezone in `infoCourse.json` from `institutions.display_timezone`.
* On sync, set `course.display_timezone = courseInfo.timezone ?? 'America/Chicago'`
* On sync, set `course_instance.display_timezone = courseInstance.timezone ?? courseTimezone`

This will be no worse than what we currently have, and will let us make all timezones NOT NULL.

Reply:
Can the default institution be created with a default timezone? This can be set to `America/Chicago` for now to maintain compatibility. We can then have some call of action asking all courses to assign a timezone in their `infoCourse.json` (e.g., a sync warning, a Slack/GitHub announcement, and a news item), and after a few months switch the default to `UTC`.

Reply:
Moved this discussion to #7344, for future action.

Reply:
I added the migration in 516b490 and updated this page to fall back to `institutions.display_timezone` in 537b211.